src/data-model-providers/codegen/ServerClusterInterfaceRegistry.h

```diff
@@ -129,9 +142,6 @@ class ServerClusterInterfaceRegistry
     /// the list is only valid as long as the registry is not modified.
     ClustersList ClustersOnEndpoint(EndpointId endpointId);
 
-    /// Unregister all registrations for the given endpoint.
-    void UnregisterAllFromEndpoint(EndpointId endpointId);
```

Comment:
OK, so what's supposed to happen when an endpoint is removed?  How is that supposed to work?

Reply:
I think we have to describe the use case and then figure out what to do. An app could in theory just list clusters on an endpoint and unregister them manually, however this does not work great if clusters return multiple endpoints (one unregister would remove items from multiple endpoints as a result).

Do we have specific applications/usages that require a convenient "remove endpoint" (maybe bridges) that we can take a look at to define the underlying requirements?

Reply:
>  think we have to describe the use case and then figure out what to do

I described the use case: an endpoint is removed/disabled.  How is that supposed to work?

> Do we have specific applications/usages that require a convenient "remove endpoint" (maybe bridges) 

100% bridges.  Some of the fabric sync stuff, iirc, for similar reasons.

But also: Matter.framework exposes APIs that allow API consumers to define endpoints, and turn them off, for whatever reasons make sense to them.  For example, maybe you have endpoints that represent the state of something that isn't always around....  So "sort of similar to bridges" in that sense, but the "something" could be all sorts of things (a mobile app, a mobile device, some random application that is not always running, etc, etc).

Reply:
Updated the restriction in the registry to require every path to share the same endpointid. This seems not ideal, however for now allows me to provide a `UnregisterAllFromEndpoint` that works.